core/server/master/src/main/java/alluxio/master/meta/ConfigRecorder.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.wire.ConfigProperty;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is responsible for containing server-side configurations.
+ */
+public class ConfigRecorder {
+  /** Stores the configuration of a master/worker with Long id. */
+  private Map<Long, List<ConfigProperty>> mConfMap;
+  /** Stores the configuration of a lost master/worker with Long id. */
+  private Map<Long, List<ConfigProperty>> mLostConfMap;
+
+  /**
+   * Constructs a new {@link ConfigRecorder}.
+   */
+  public ConfigRecorder() {
+    mConfMap = new HashMap<>();
+    mLostConfMap = new HashMap<>();
+  }
+
+  /**
+   * Registers new configuration information.
+   *
+   * @param id the master/worker id
+   * @param configList the configuration of this master/worker
+   */
+  public synchronized void registerNewConf(long id, List<ConfigProperty> configList) {
+    mConfMap.put(id, configList);
+    mLostConfMap.remove(id);
+  }
+
+  /**
+   * Removes the configuration of a master/worker from ConfMap to LostConfMap
```

Comment:
ConfMap and LostConfMap are implementation details that don't need to be exposed to users of the class, they shouldn't be part of the javadoc